From 28c54f3d551af09c414fbf2e2e13a9f7b61781f9 Mon Sep 17 00:00:00 2001
From: Jonathan Kang <jonathan121537@gmail.com>
Date: Tue, 5 Apr 2016 14:15:20 +0800
Subject: [PATCH] network: Fix empty Wifi list

Add a callback function for signals "access-point-added" and
"access-point-removed" to update the Wifi list.

https://bugzilla.gnome.org/show_bug.cgi?id=709641
---
 panels/network/net-device-wifi.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/panels/network/net-device-wifi.c b/panels/network/net-device-wifi.c
index e7d0c93..ca55d94 100644
--- a/panels/network/net-device-wifi.c
+++ b/panels/network/net-device-wifi.c
@@ -240,6 +240,18 @@ get_ap_security_string (NMAccessPoint *ap)
 }
 
 static void
+net_device_wifi_access_point_changed (NMDeviceWifi *nm_device_wifi,
+                                      NMAccessPoint *ap,
+                                      gpointer user_data)
+{
+        NetDeviceWifi *device_wifi;
+
+        device_wifi = NET_DEVICE_WIFI (user_data);
+
+        populate_ap_list (device_wifi);
+}
+
+static void
 wireless_enabled_toggled (NMClient       *client,
                           GParamSpec     *pspec,
                           NetDeviceWifi *device_wifi)
@@ -1317,6 +1329,13 @@ net_device_wifi_constructed (GObject *object)
 
         nm_device = net_device_get_nm_device (NET_DEVICE (device_wifi));
 
+        g_signal_connect_object (nm_device, "access-point-added",
+                                 G_CALLBACK (net_device_wifi_access_point_changed),
+                                 device_wifi, 0);
+        g_signal_connect_object (nm_device, "access-point-removed",
+                                 G_CALLBACK (net_device_wifi_access_point_changed),
+                                 device_wifi, 0);
+
         /* only enable the button if the user can create a hotspot */
         widget = GTK_WIDGET (gtk_builder_get_object (device_wifi->priv->builder,
                                                      "start_hotspot_button"));
-- 
2.7.4

